### ISSUE TITLE
v2.0.0 crashes on startup

### ISSUE THREAD

**Issue by gpbenton** [ostrya/PresencePublisher#21]
It says I must enable location services (which wasn't necessary before) and then crashes.  I would add some logs, but it doesn't stay up long enough to get any.
Sony Z5 Compact - Android 5.1.1

**Comment by ostrya** [comment]
In Android less than version 9, the location permission and the location services are only necessary if you want to use bluetooth beacons. Does it also crash if you cancel the dialog?

**Comment by gpbenton** [comment]
Unfortunately, yes it does.

If I press OK, it goes to the android location properties gui and crashes.

If I press Cancel it goes to the PresencePublisher gui  and crashes.

**Comment by ostrya** [comment]
Can you try with the following debug build? I adapted the app to send the stack trace via send intent, so you should be able to open it e.g. in a mail app.

[app-debug.zip](https://github.com/ostrya/PresencePublisher/files/4155275/app-debug.zip)

**Comment by ostrya** [comment]
@gpbenton If you still see the issue with a more recent release, can you please post a stacktrace? Otherwise I will close this issue.

**Comment by gpbenton** [comment]
I'll give a later release a try when I get a chance.  Upgrading and then downgrading is quite a bit of  work though, and the old release does exactly what I need...

**Comment by ostrya** [comment]
Thanks for that. I have attached a new debug build with a different package name here
[app-debug.zip](https://github.com/ostrya/PresencePublisher/files/5338456/app-debug.zip)
, so you can safely install it in parallel to your existing app.

**Comment by hamstie** [comment]
Hi Kai,i tried to install the app-debug.apk. Android 9 said: Nothing installed at the end of the process.


    Am Mittwoch, 7. Oktober 2020, 07:45:22 MESZ hat Kai Helbig <notifications@github.com> Folgendes geschrieben:  
 
 


Thanks for that. I have attached a new debug build with a different package name here app-debug.zip, so you can safely install it in parallel to your existing app.

**Comment by ostrya** [comment]
Thanks for the feedback. I again seem to have picked a file built incrementally via IDE rather than from scratch via Gradle. New file is attached here [app-debug.zip](https://github.com/ostrya/PresencePublisher/files/5339580/app-debug.zip).

**Comment by hamstie** [comment]
i wasn't able to extract the app-debug.apk archive, that was the problem. 
md5sum : 6d28cfd0960c5a93a279bce5c548fc2 

the new apk is fine, i will testing.......

Thank you


    Am Mittwoch, 7. Oktober 2020, 11:38:07 MESZ hat Kai Helbig <notifications@github.com> Folgendes geschrieben:  
 
 


Thanks for the feedback. I again seem to have picked a file built incrementally via IDE rather than from scratch via Gradle. New file is attached here app-debug.zip.

**Comment by gpbenton** [comment]
I also downloaded the apk and it started up fine.  I haven't connected to the broker yet, because its late and I have been drinking beer.
I did press the bluetooth button, and that crashed the app, but that's a different problem, and I don't have any bluetooth beacons anyway.
But does this mean if I upgrade from FDroid I should be able to get up and running again without having to uninstall, reinstall, configure my mqtt connection and reapply my network connection settings?

**Comment by hamstie** [comment]
Hi Kai,i was tested some things in the past two days (V2.2.0):- Bluetooth search no, exception- retain: working fine- QoS not tested- send cycle, looks stable

One very very small hint: I wonder who was set a test=test topic in my mqtt-tree, remember it is retained and will be "never" lost or changed. Perhaps it is better: - don't flag it retained,or- change the payload for each test, and add some useful things:  send-timestamp, test counter,  app-name with version
- give the use a hint in the UI. ....published to topic:test....
Thank you for your very nice appregards 
  Mark




    Am Mittwoch, 7. Oktober 2020, 23:30:57 MESZ hat gpbenton <notifications@github.com> Folgendes geschrieben:  
 
 


I also downloaded the apk and it started up fine. I haven't connected to the broker yet, because its late and I have been drinking beer.
I did press the bluetooth button, and that crashed the app, but that's a different problem, and I don't have any bluetooth beacons anyway.
But does this mean if I upgrade from FDroid I should be able to get up and running again without having to uninstall, reinstall, configure my mqtt connection and reapply my network connection settings?
